### PR TITLE
Update Elasticsearch docs to describe incoming message types

### PR DESCRIPTION
**Pull request by takezoe**
This pull request updates Elasticsearch document according to #1104.

Also this pull request contains a bugfix for Java API. I found the bug when I wrote a test for Java API to post an example in the document.

### CODE REVIEW

File: docs/src/main/paradox/elasticsearch.md [hunk 1]
```diff
@@ -66,6 +66,23 @@ Java
 : @@snip ($alpakka$/elasticsearch/src/test/java/akka/stream/alpakka/elasticsearch/ElasticsearchTest.java) { #run-jsobject }
 
 
+### Incoming message types
+
+In the above examples, `IncomingIndexMessage` is used as the input to `ElasticsearchSink`. This means requesting `index` operation to Elasticsearch. It's able to request other operations using following message types:
```
Comment (2m) [inline docs/src/main/paradox/elasticsearch.md]: s/able/possible

File: elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchFlowStage.scala [hunk 2]
```diff
@@ -57,7 +57,7 @@ object IncomingUpsertMessage {
 
   // Java-api - without passThrough
   def create[T](id: String, source: T): IncomingMessage[T, NotUsed] =
-    IncomingUpdateMessage(id, source)
+    IncomingUpsertMessage(id, source)
```
Comment (2m) [inline elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchFlowStage.scala]: Ahh. Nice catch. :)

File: docs/src/main/paradox/elasticsearch.md (hunk 1 above)
Comment (takezoe) [reply docs/src/main/paradox/elasticsearch.md]: Fixed.

File: elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchFlowStage.scala (hunk 2 above)
Comment (takezoe) [reply elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchFlowStage.scala]: I should have written also Java test from the beginning...

**Pull request merged**